torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -858,6 +1043,21 @@ void ProcessReshapeNode(Node* n, int opset_version) {
     }
   }
 
+  if (ConstantValueMap::HasShapeValue(n->input(1)->debugName())) {
+    auto shape_vector_1 =
+        ConstantValueMap::GetShapeValue(n->input(1)->debugName()).value();
+    if (ConstantValueMap::HasShape(n->input(0)->debugName())) {
+      auto shape_vector_0 =
+          ConstantValueMap::GetShape(n->input(0)->debugName()).value();
+      auto final_shape = ComputeShapeFromReshape(
+          n, shape_vector_0, shape_vector_1, opset_version);
+      if (final_shape.has_value()) {
+        UpdateShape(n->output(), final_shape.value());
+        return;
+      }
+    }
+  }
+
   if (ConstantValueMap::HasShape(n->input(1)->debugName())) {
```

Comment:
ditto here on `else if`?

Reply:
No. When the early return does not happen, it will go to this logic.